app/scripts/views/sign_up.js

```diff
@@ -295,33 +319,13 @@ define(function (require, exports, module) {
       }
 
       return self.invokeBrokerMethod('beforeSignIn', account.get('email'))
-        .then(function () {
-          return self.user.signUpAccount(account, password, self.relier, {
-            resume: self.getStringifiedResumeToken()
-          });
-        })
-        .then(function (account) {
-          // formPrefill information is no longer needed after the user
-          // has successfully signed up. Clear the info to ensure
-          // passwords aren't sticking around in memory.
-          self._formPrefill.clear();
-
-          if (preVerifyToken && account.get('verified')) {
-            self.logViewEvent('preverified.success');
-          }
-          self.logViewEvent('success');
-          return self.invokeBrokerMethod('afterSignUp', account);
-        })
-        .then(self.onSignUpSuccess.bind(self, account))
-        .fail(self.signUpError.bind(self));
+        .then(action.bind(self, account, password, preVerifyToken));
     },
 
-    signUpError: function (err) {
+    onSignUpError: function (err) {
       var self = this;
-      // Account already exists. No attempt is made at signing the
-      // user in directly, instead, point the user to the signin page
-      // where the entered email/password will be prefilled.
-      if (AuthErrors.is(err, 'ACCOUNT_ALREADY_EXISTS')) {
+      if (AuthErrors.is(err, 'INCORRECT_PASSWORD')) {
+        // Account already exists, sign-in was attempted but password was wrong.
         return self._suggestSignIn(err);
```

Comment:
@ryanfeeley's request for [incorrect password](https://github.com/mozilla/fxa-content-server/issues/2778#issuecomment-163380724) is slightly different:

> When the user enters valid username but invalid password they are taken to login page and shown "Incorrect password" error message.

We'll have to redirect the user to `/signin` and show them the `INCORRECT_PASSWORD` error on that screen.